src/ir/utils.rs

```diff
@@ -408,9 +408,9 @@ where
     C: Ctx<T>,
 {
     /// A reference to the underlying value.
-    key: Idx<T>,
+    pub key: Idx<T>,
     /// A reference to the owner of the foreign key.
-    owner: Idx<C>,
+    pub owner: Idx<C>,
```

Comment:
Making these `pub` will allow for mutable access to these fields which we want to avoid. I think you can either define a `take()` method that consumes the `Foreign` and returns the two fields or define accessor methods that don't give mutable access to these fields.

Reply:
This is done in bundle elim by the way in case that gets merged.

Reply:
Seems like `bundle-elim` no longer does this